### PR TITLE
feat: add GitLab connector

### PR DESCRIPTION
**Pull request by devtekve**
## Summary

This MR introduces the initial version of the GitLab connector for Logto, allowing users to authenticate via GitLab accounts. The implementation is based on the existing Patreon connector, adapting it to use GitLab's OAuth 2.0 authentication flow.

### Key Changes:
- Added a new `@logto/connector-gitlab` package.
- Implemented OAuth 2.0 authentication for GitLab.
- Configured endpoints for fetching user details and handling tokens specific to GitLab.
- Updated README to reflect setup and configuration instructions for the GitLab connector.

## Testing

- Created unit tests to verify the OAuth token exchange process.
- Added integration tests to ensure the connector correctly interacts with GitLab's API.
- Manually tested the connector by authenticating with a real GitLab account, verifying that user details are retrieved correctly.
- Verified that errors are handled gracefully, including invalid credentials and denied access.

## Checklist

- [x] `.changeset` - Added a changeset file to record the new feature introduction.
- [x] unit tests - Developed unit tests for OAuth token exchange and user detail retrieval.
- [x] necessary TSDoc comments - Added TSDoc comments to the new methods and classes introduced for documentation.
- [ ] integration tests - Ran integration tests to ensure proper interaction

For maintainers: This code is yours, feel free to update and modify as you see fit.

### CODE REVIEW

**Comment by devtekve** [comment]
_**NOTE TO MAINTAINERS**_
Do not merge this _yet_ there's some clarifications I need to do. Details here https://discord.com/channels/965845662535147551/1278449432500768841/1278893773127942198

**Comment by devtekve** [comment]
Ready to re-review, I've updated the documentation, the default scopes and the user info response guard to align with the minimum required set of variables expected

File: packages/connectors/connector-gitlab/src/index.ts
Comment (xiaoyijun) [inline packages/connectors/connector-gitlab/src/index.ts:148]: We will not return the `email` if the `email_verfied` is `false`, since it will be used as the primary email of the user.

File: packages/connectors/connector-gitlab/src/constant.ts
Comment (xiaoyijun) [inline packages/connectors/connector-gitlab/src/constant.ts:29]: This placeholder seems unusual, we can align with other connectors.

**Review by xiaoyijun — COMMENTED** [review]
Overall looks good. However, some logics should be updated. @devtekve

File: packages/connectors/connector-gitlab/src/index.ts
Comment (devtekve) [reply packages/connectors/connector-gitlab/src/index.ts:148]: Makes total sense

File: packages/connectors/connector-gitlab/src/constant.ts
Comment (devtekve) [reply packages/connectors/connector-gitlab/src/constant.ts:29]: I was hesitant on it for that same reason, I’ll remove it and leave it as it is on other connectors

Comment (devtekve) [reply packages/connectors/connector-gitlab/src/constant.ts]: Done, I've removed this line and left the default 
image

File: packages/connectors/connector-gitlab/src/index.ts
Comment (devtekve) [reply packages/connectors/connector-gitlab/src/index.ts]: done

File: packages/connectors/connector-gitlab/src/index.ts
Comment (darcyYe) [inline packages/connectors/connector-gitlab/src/index.ts:154]:
```suggestion
      return {
        id,
        avatar,
        name,
        ...conditional(email_verified && { email }),
        rawData,
      };
```
Can refer to [implementation on Discord connector](https://github.com/logto-io/logto/blob/57d97a4df82c6a5a7132c4cc3b978557b8d5c308/packages/connectors/connector-discord/src/index.ts#L127).

**Review by darcyYe — APPROVED** [review]
overall LGTM

File: packages/connectors/connector-gitlab/src/index.ts
Comment (devtekve) [reply packages/connectors/connector-gitlab/src/index.ts:154]: HAAAA!!!! Thank you very much for this one!!!!! I was like "I don't really like much the ternary here" hahahaha

File: packages/connectors/connector-gitlab/src/index.ts
Comment (devtekve) [reply packages/connectors/connector-gitlab/src/index.ts:52]: makes sense to me. I wanted to highlight the "enforcement" of that scope but I think I can do it in a different more maintainable way

File: packages/connectors/connector-gitlab/src/index.ts
Comment (devtekve) [reply packages/connectors/connector-gitlab/src/index.ts]: done

File: packages/connectors/connector-gitlab/src/index.ts
Comment (devtekve) [reply packages/connectors/connector-gitlab/src/index.ts]: Addressing in the other comment

Comment (devtekve) [reply packages/connectors/connector-gitlab/src/index.ts]: @darcyYe I updated this. What I did is not _**exactly**_ your suggestion, so I am leaving this "unresolved" to give you the final saying on it. If you want me to change it to just having one property `defaultScope` and have the  `openid` there I can do that too. No strong opinions on it. I just wanted to make sure it was clearly conveyed that `openid` is not overrideable, it's required. While allowing for future expansion with other "default scopes" than could be overriden.

**Review by xiaoyijun — APPROVED** [review]
LGTM

**Review by gao-sun — APPROVED** [review]
changeset lgtm

File: packages/connectors/connector-gitlab/src/index.ts
Comment (darcyYe) [reply packages/connectors/connector-gitlab/src/index.ts]: The latest implementation looks good to me.

**Pull request merged**
